crates/astria-conductor/src/data_availability.rs

```diff
@@ -148,9 +159,10 @@ impl Reader {
     }
 
     fn get_latest_height(&mut self) {
-        let client = self.celestia_client.inner().clone();
+        use celestia_client::celestia_rpc::HeaderClient;
+        let client = self.celestia_client.clone();
         self.get_latest_height = Some(tokio::spawn(async move {
-            Ok(client.header_network_head().await?.height())
+            Ok(client.header_network_head().await?.header.height.value())
```

Comment:
I moved everything from u64 to (celestia-tendermint) Height because the use of `Head::value` everywhere was indeed getting silly.

The extension trait also saw some changes to take height arguments as `T: Into<u64> + Send` instead so that it becomes convenient to use with `Height`.